crates/services/compression/src/ports/compression_storage.rs

```diff
@@ -61,10 +63,27 @@ where
         compressed_block: &CompressedBlock,
     ) -> crate::Result<usize> {
         let height = (*height).into();
+
         self.storage_as_mut::<CompressedBlocks>()
             .insert(&height, compressed_block)
             .map_err(CompressionError::FailedToWriteCompressedBlock)?;
 
+        #[cfg(feature = "fault-proving")]
+        {
+            use crate::temporal_registry::fault_proving::ComputeRegistryRoot;
+            use storage::{
+                compressed_blocks::CompressedBlocks,
+                CompressedBlocksRoots,
+            };
+
+            let root =
+                <Self as ComputeRegistryRoot>::root_of_table::<CompressedBlocks>(self)
+                    .map_err(CompressionError::FailedToComputeCompressedBlockRoot)?;
+            self.storage_as_mut::<CompressedBlocksRoots>()
+                .insert(&height, &root)
+                .map_err(CompressionError::FailedToWriteCompressedBlockRoot)?;
+        };
```

Comment:
I don't think that we need an additional `CompressedBlocksRoots` table. If  `CompressedBlocks` uses `Sparse` blueprint, the nit shoudl be enought for us to get the root.